crates/router/src/core/payments/operations/payment_create.rs

```diff
@@ -576,6 +585,22 @@ impl PaymentCreate {
             encoded_data: None,
         }
     }
+
+    #[instrument(skip_all)]
+    pub async fn create_ephemeral_key(
+        &self,
+        state: &AppState,
+        customer_id: String,
+        merchant_id: String,
+    ) -> Option<ephemeral_key::EphemeralKey> {
+        let ephemeral_key: Option<ephemeral_key::EphemeralKey> =
+            match helpers::make_ephemeral_key(state, customer_id, merchant_id).await {
+                Ok(services::ApplicationResponse::Json(ek)) => Some(ek),
+                Ok(_) => None,
+                Err(_) => None,
+            };
+        ephemeral_key
+    }
```

Comment:
I feel this function is redundant. You can just do `helpers::make_ephemeral_key(state,customer_id,merchant_id).ok()` in https://github.com/juspay/hyperswitch/blob/801f356fccdc189a94489669a84ea5ac77178f57/crates/router/src/core/payments/operations/payment_create.rs#L55

Reply:
`helpers::make_ephemeral_key` returns a `Future<CustomResult<services::ApplicationResponse<EphemeralKey>>, ApiErrorResponse>`

I thought of moving all the matching logic to this function `PaymentCreate::create_ephemeral_key()`. 
If I call `helpers::make_ephemeral_key` inside `get_trackers()`, then I will have to add 6 more lines inside `get_trackers()`

Reply:
You can do something like
```rust
helpers::make_ephemeral_key(/*args*/).await.ok().and_then(|ek| 
if let services::ApplicationResponse::json(ek) = ek {
	Some(ek)
} 
else {
	None
});
```
Don't think we need a seperate function here